WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift

```diff
@@ -460,19 +465,23 @@ extension LoginEmailViewController {
         loginFields.nonceUserID = userID
 
         performSegue(withIdentifier: NUXAbstractViewController.SegueIdentifier.show2FA, sender: self)
+        WPAppAnalytics.track(.loginSocial2faNeeded)
     }
 }
 
 extension LoginEmailViewController: GIDSignInDelegate {
     func sign(_ signIn: GIDSignIn?, didSignInFor user: GIDGoogleUser?, withError error: Error?) {
-        // TODO: finish implementing wpcom login via Google code
         guard let user = user,
-              let token = user.authentication.idToken,
-              let email = user.profile.email else {
-            // The Google SignIn for may have been canceled.
-            //TODO: Add analytis
-            configureViewLoading(false)
-            return
+            let token = user.authentication.idToken,
+            let email = user.profile.email else {
+                // The Google SignIn for may have been canceled.
+                if let err = error {
+                    WPAppAnalytics.track(.loginSocialButtonFailure, error: err)
+                } else {
+                    WPAppAnalytics.track(.loginSocialButtonFailure)
```

Comment:
This will track a failure by dismissing either the permission alert or Google login screens. Is that intentional?

Reply:
It is.  The documentation of the corresponding event on the web mentions that it tracks just dismissing the popup modal.  I wired it this way for consistency.